exchanges/gateio/gateio.go

```diff
@@ -175,13 +175,13 @@ var validTimesInForces = []struct {
 	{gtcTIF, order.GoodTillCancel}, {iocTIF, order.ImmediateOrCancel}, {pocTIF, order.PostOnly}, {fokTIF, order.FillOrKill},
 }
 
-func timeInForceFromString(tif string) order.TimeInForce {
+func timeInForceFromString(tif string) (order.TimeInForce, error) {
 	for a := range validTimesInForces {
 		if validTimesInForces[a].String == tif {
-			return validTimesInForces[a].TimeInForce
+			return validTimesInForces[a].TimeInForce, nil
 		}
 	}
-	return order.UnknownTIF
+	return order.UnknownTIF, fmt.Errorf("%w: %q", order.ErrUnsupportedTimeInForce, tif)
```

Comment:
🐒 I think `Unsupported` isn't the right term. It's not "not supported". It's not valid.
```suggestion
	return order.UnknownTIF, fmt.Errorf("%w: %q", order.ErrInvalidTimeInForce, tif)
```

Reply:
But we are matching among the allowed possible timeInForce values. not from all time-in-force values supported by the exchange.

Reply:
Sorry, you're right 👍